Mecanum用のclassを設定した

hs4449889 opened the pull request:
# 変更の目的
メカナムホイール用のclassを作成した。

-------------------------------------
# 関連issue
#26 

--------------------------------------

# レビュー完了希望日

### 2021/10/6 (水) 


# After merge
Fix #26 

--------------------------------------

hs4449889 committed:
[upd]controller classを更新した

・レバーもR2,L2も最高値を取るときのみ検知する設定にしたので実質ボタン
計23個のボタンが有る
・値設定しんどいがな~~~

hs4449889 committed:
[add]Controller classを追加した(値追記)

Ps4コントローラーの条件文の値も設定した(締切は守った)
・RLレバーどちらも先輩の値と12/14のまとめ表の値が違っていたので要確認
(新しい12/14の方に合わせておいた)
・また、レバーを押し込んだときの値も確認できていないので金研で値を出してまとめる

hs4449889 committed:
[add]Mecanum用classの作成

・インスタンスの変数を設定した
・class全体を説明するdocstringsを書いた
・オブジェクト内で動作を支持するspecify_motion という変数を保有するclass
・呼び出し時の引数がやや見ずらい気もするが、余裕があれば改善する

hs4449889 committed:
[upd]Mecanumクラスのメカナムホイールの動作を指定する関数を追加した

動作を指定するspecify_motionの値を変更するだけの関数を追加したが
mainを書くときの可読性が上がると判断したためこのままで行く

hs4449889 committed:
[upd]mecanum_stop関数を追加した。

・動作指示用の関数の値設定は必要性がないと設定する意欲がわかないのでので省略します。

HS-1114G commented on Mecanum/mecanum.py:
```diff
@@ -0,0 +1,74 @@
+
+    """
+    #// to do //配列にまとめるかどうか相談
+    #最終的に辞書を採用(配列は可読性が低い)
```
辞書あんまりちゃんと知らないから、何とも言えない…
とりあえずめんどくさいことだけは知ってるから、できれば使いたくないな…

hs4449889 replied:
添字じゃなくてkeyでアクセスするから可読性は上がると思ふ

HS-1114G commented on Mecanum/mecanum.py:
```diff
@@ -0,0 +1,74 @@
+
+    #値設定は後回しで
+    def left_translation(self):
+        self.specify_motion[,,,]
```
self.specify_motion = [1, 2, 3, 4]
じゃない？

hs4449889 replied:
値の設定をサボったやつなので、暇なときとかに更新します

HS-1114G commented on Controller/ps4.py:
```diff
@@ -0,0 +1,189 @@
+#PS4　class
+
+#TO DO : valueが可変するボタンに対応する
```
このファイルって、このプルリクで直した奴じゃないよね？？
このファイル更新したら、前みたいにめんどいことになるんじゃない？？

HS-1114G reviewed (changes requested):
メカナムの方向指定と、その他もろもろよろしくお願いします。

hs4449889 committed:
[upd]Controllerクラスのインスタンス変数としてps4の各ボタンの値を設定した

・ループの最初に毎回このps4オブジェクトを呼び出し、
　そのループのときのps4からの値をインスタンス変数とすることで確実に使えるようにした

この改善策で懸念されるのは、pythonはオブジェクトで使ったメモリを自分で解放できない？
ということが問題として上がるかもしれないということ(ここは後でよく調べておく)

hs4449889 committed:
[upd]メカナムホイールの動作指定をするインスタンス変数specify_motionの値を設定した

・4方向+左側にターン+右側にターン

hs4449889 committed:
[sytle]specify_motionの変更をちょっと見やすくした

・-1以外前と後ろに空白1ついれる
・-1は後ろだけ空白を1つ入れる

HS-1114G commented on Mecanum/mecanum.py:
```diff
@@ -0,0 +1,74 @@
+        self.rear_left   = dc_motor.DcMotor(pwm_pin = rear_left["pwm_pin"],dir_pin= rear_left["dir_pin"])
+        self.rear_right  = dc_motor.DcMotor(pwm_pin = rear_right["pwm_pin"],dir_pin= rear_right["dir_pin"])
+
+        self.specify_motion = [-1 ,-1 ,-1 ,-1 ]
```
ここに「＝」ついてるよね？？
※下のコメントの参考

HS-1114G commented on Mecanum/mecanum.py:
```diff
@@ -0,0 +1,74 @@
+
+    #値設定は後回しで
+    def left_translation(self):
+        self.specify_motion[ 0 , 1 , 1 , 0 ]
```
self.specify_motion = [ 0, 1, 1, 0]
じゃない？
※これ以下すべてを編集した方がいい気がする…

HS-1114G reviewed (changes requested):
おそらく下の配列の定義は直さなければいけないと思います。
また、複数のファイルが同時に更新されているようですが大丈夫なのですか？
確認よろしくお願いいたします。
